### ISSUE TITLE
loss value = global step?

### ISSUE THREAD

**Issue by auzxb** [tensorflow/magenta#853]
Hi, I meet the problem that the # loss value is equal to the # global step?

INFO:tensorflow:global step 500: loss = 500.0000 (1.11 sec/step)

Is this a bug , or I run the script with some mistakes?

Thank you!

**Comment by auzxb** [comment]
I run the nsynth/wavenet/train.py script

**Comment by auzxb** [comment]
I run the nsynth/wavenet/train.py script.  @jesseengel @ @adarob

**Comment by auzxb** [comment]
I run the nsynth/wavenet/train.py script. Thank you! @jesseengel @ @adarob

**Comment by auzxb** [comment]
I modifed the code  and solve the problem(#loss value = #global step). However, I got a nan loss after few steps of trainning with the default parameters.
 train_op = slim.learning.create_train_op(total_loss = total_loss, 
                                               optimizer = opt,
                                               global_step = global_step,
                                               colocate_gradients_with_ops=False)

**Comment by auzxb** [comment]
I modifed the code and solve the problem(#loss value = #global step). However, I got a nan loss after few steps of trainning with the default parameters.
```
train_op = slim.learning.create_train_op(total_loss = total_loss,
optimizer = opt,
global_step = global_step,
colocate_gradients_with_ops=False)
```

**Comment by auzxb** [comment]
I modifed the code and solve the problem(#loss value = #global step). However, I got a nan loss after few steps of trainning with the default parameters.
train_op = slim.learning.create_train_op(total_loss = total_loss,
optimizer = opt,
global_step = global_step,
colocate_gradients_with_ops=True)

**Comment by auzxb** [comment]
I modifed the code and solve the problem(#loss value = #global step). However, I got a nan loss after few steps of trainning with the default parameters.
```
train_op = slim.learning.create_train_op(total_loss = total_loss,
optimizer = opt,
global_step = global_step,
colocate_gradients_with_ops=True)
```

**Comment by auzxb** [comment]
I fix the bug！
The audio samples are represented as floating point values in the range [-1,1].
Audio point values will be quantized to [0,256] using the mu_law method.
Howerve, the range of [0,255] is expected for tf.nn.sparse_softmax_cross_entropy_with_logits method.
Thus, scale the audio into [-1,1) will avoid the above bug.

```
def mu_law(x, mu=255, int8=False):
  """A TF implementation of Mu-Law encoding.

  Args:
    x: The audio samples to encode.
    mu: The Mu to use in our Mu-Law.
    int8: Use int8 encoding.

  Returns:
    out: The Mu-Law encoded int8 data.
  """
  out = tf.sign(x) * tf.log(1 + mu * tf.abs(x)) / np.log(1 + mu)
  out = tf.floor(out * 128)
  if int8:
    out = tf.cast(out, tf.int8)
  return out
```

**Comment by jesseengel** [comment]
Thanks for looking into this. Would you be willing to submit a small PR with your fixes?

**Comment by 493238731** [comment]
@auzxb Can you show how to solve the `NaN loss` bug more detail? Thanks

**Comment by auzxb** [comment]
Just scale the audio into [-1,1) instead of [-1, 1] can avoid the NaN loss.

**Comment by 493238731** [comment]
How to modify the code?
